xonsh/tools.py

```diff
@@ -319,11 +318,10 @@ def suggest_commands(cmd, env, aliases):
                 suggested[a] = 'Alias'
 
     for d in filter(os.path.isdir, env.get('PATH')):
-        for f in os.listdir(d):
-            if f not in suggested:
-                if levenshtein(f.lower(), cmd, thresh) < thresh:
-                    fname = os.path.join(d, f)
-                    suggested[f] = 'Command ({0})'.format(fname)
+        for name, path in ((x.name, x.path) for x in scandir(d)
+                           if x.name not in suggested):
```

Comment:
No reason to double loop here.  Please just make a single loop.  Loops in Python are super slow.